src/editor/mod.rs

```diff
@@ -37,52 +37,75 @@ pub struct Editor {
     pub cursor: Cursor,
     pub default_style: Arc<Style>,
     pub defined_styles: HashMap<u64, Arc<Style>>,
-    pub previous_style: Option<Arc<Style>>
+    pub previous_style: Option<Arc<Style>>,
 }
 
 impl Editor {
     pub fn new() -> Editor {
         let mut editor = Editor {
-            grid: CharacterGrid::new(INITIAL_DIMENSIONS),
+            grid: CharacterGrid::new(get_initial_dimensions()),
             title: "Neovide".to_string(),
             font_name: None,
             font_size: None,
             cursor: Cursor::new(),
-            default_style: Arc::new(Style::new(Colors::new(Some(colors::WHITE), Some(colors::BLACK), Some(colors::GREY)))),
+            default_style: Arc::new(Style::new(Colors::new(
+                Some(colors::WHITE),
+                Some(colors::BLACK),
+                Some(colors::GREY),
+            ))),
             defined_styles: HashMap::new(),
-            previous_style: None
+            previous_style: None,
         };
 
         editor.grid.clear();
         editor
     }
 
-
     pub fn handle_redraw_event(&mut self, event: RedrawEvent) {
         match event {
             RedrawEvent::SetTitle { title } => self.title = title,
             RedrawEvent::ModeInfoSet { cursor_modes } => self.cursor.mode_list = cursor_modes,
             RedrawEvent::OptionSet { gui_option } => self.set_option(gui_option),
-            RedrawEvent::ModeChange { mode_index } => self.cursor.change_mode(mode_index, &self.defined_styles),
+            RedrawEvent::ModeChange { mode_index } => {
+                self.cursor.change_mode(mode_index, &self.defined_styles)
+            }
             RedrawEvent::BusyStart => {
                 trace!("Cursor off");
                 self.cursor.enabled = false;
-            },
```

Comment:
I assume most of the changes on this file are from rustfmt?

Reply:
I can't tell if there are any functional changes in this diff. Maybe revert any formatting changes? I'm interested in adding rustfmt globally at some point, but doing it on only one file seems distracting for this PR

Reply:
Sorry about that. I started working on this before it crossed my mind about whether or not I would even be sharing this. I'm reverting the formatting for most of the file. The only major change is that I switched out the global variable for a getter.